core/sql/common/str.cpp

```diff
@@ -997,19 +997,40 @@ Lng32 str_decode(void *tgt, Lng32 tgtMaxLen, const char *src, Lng32 srcLen)
   return length;
 }
 
-// Strips leading and trailing blanks. src will contain a NULL after the
+// Strips leading and/or trailing blanks. src will contain a NULL after the
 // end of the first non-blank character.The length of the "stripped" string
-// is returned in len
-
-void str_strip_blanks(char *src , Lng32 &len)
+// is returned in len.
+// Returns pointer to the start of string after leading blanks.
+char * str_strip_blanks(char *src , Lng32 &len, 
+                        NABoolean stripLeading,
+                        NABoolean stripTrailing
+                        )
 {
+  if (! src)
+    return NULL;
+
   len = str_len(src)-1;
-  while ((len >= 0) && (src[len] == ' '))
-    len--;
+  if (len <= 0)
+    return src;
```

Comment:
Hmmm... Isn't len a reference parameter? Therefore it is an output as well as an input. Perhaps it should be pass by value instead?

Let's take another example. Suppose the input string is ' ' (that is, one blank). The code as written would calculate len = 0 at line 1012, then the 'if' at line 1013 is true, and we return src without stripping a blank. Seems like we should return the empty string instead?